### ISSUE TITLE
"Expand macro recursively" on builtin derive shows an empty impl block

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#14235]
For example:

```rust
#[derive(Clone, Debug)]
pub struct SharedObservableBase<L>(pub L);
```

Putting the cursor on either of the derives and running "Expand macro recursively" gives me sth. like this:

```rust
// Recursive expansion of Clone! macro
// ====================================

impl<T0: core::clone::Clone> core::clone::Clone for SharedObservableBase<T0> {}
```

**rust-analyzer version**: `0.3.1418-standalone`

**rustc version**: `1.67.0 (fc594f156 2023-01-24)`

**Comment by Veykril** [comment]
The expansions for those builtins are dummies currently

**Comment by jplatte** [comment]
Right. Maybe return an error when attempting to expand a builtin macro then? Or output another comment inside the existing output that builtin derives aren't properly supported yet?

**Comment by Veykril** [comment]
We should just emit something sensible there (once inline macro works with derives it will benefit from that as well)

**Comment by jplatte** [comment]
You mean RA should re-implement rustc's derives entirely? That seems a little worrying if the output can be different between rustc and RA (for example if new attributes akin to `#[default]` are added), I would never suspect RA's "expand macro recursively" output to be possibly different from what the compiler expands. (of course regular text expansion not showing reused hygienic identifiers already screws with this, but only in edge cases and that should be fixable as well)

**Comment by Veykril** [comment]
~ there is nothing that specifies the exact output a compiler has to expand builtin derives to, so I don't think that's a problem. It would of course be ideal if we could re-use the backing implementation of these from rustc itself but I doubt we can easily move towards that (at least for the time being).

**Comment by jplatte** [comment]
Hm, alright. Maybe I'm too paranoid about a reimplementation producing different output. Many of them are luckily super straight-forward (I would expect, haven't looked at the impls).
